neuraxle/pipeline.py

```diff
@@ -273,92 +120,153 @@ def fit(self, data_inputs, expected_outputs=None) -> 'Pipeline':
             expected_outputs=expected_outputs
         )
 
-        new_self, _ = self._fit_transform_core(data_container)
+        context = ExecutionContext.from_root(self, self.cache_folder)
```

Comment:
Hey, is this a good idea to replace your line of code that creates the context: 
```
_, context = self.setup(context=None)
```

And in setup, we do `ExecutionContext.from_root(self, self.cache_folder)` and if we have substeps, we pass them the new context at creation and this context is pushed/poped accordingly from this point on. So whenever the context is None, it creates a default context and push things accordingly. 

The only thing I don't like about what I hereby suggest is that it makes the context not pure anymore. I'd like it to be computer as much as possible on-the-fly with the handle methods and never set too much in advance. 

So I don't know, we'll need to talk about that. The PR is fine like it is already for now.

Reply:
won't fix

Reply:
removed context from setup after discussion